sdk/search/Azure.Search/src/autorest.md

```diff
@@ -139,3 +68,30 @@ directive:
 ## C# Customizations
 Shape the swagger APIs to produce the best C# API possible.  We can consider
 fixing these in the swagger files if they would benefit other languages.
+
+### Add documentation
+Add documentation to nodes that do not have it. These should be fixed in swagger.
+``` yaml
+directive:
+- from: swagger-document
+  where: $.definitions..properties["@odata.type"]
+  transform: $.description = "The model type.";
+- from: swagger-document
+  where: $.definitions.CognitiveServicesAccount.properties.description
+  transform: $.description = "Description of the cognitive resource attached to a skillset.";
+- from: swagger-document
+  where: $.definitions.CognitiveServicesAccountKey.properties.key
+  transform: $.description = "The key used to provision a cognitive resource attached to a skillset.";
+- from: swagger-document
+  where: $.definitions.ScoringFunction.properties.type
+  transform: $.description = "Required for scoring functions. Indicates the type of function to use. Valid values include magnitude, freshness, distance, and tag. You can include more than one function in each scoring profile. The function name must be lower case.";
+```
+
+### Property name changes
+Change the name of some properties so they are properly CamelCased.
+``` yaml
+modelerfour:
+  naming:
+    override:
+      "@odata.type": ODataType
```

Comment:
Where does this get exposed out of curiosity?  Github doesn't make the generated code easy to quickly search.

Reply:
It shouldn't be exposed anywhere. `@odata.type` should be used only as a polymorphic discriminator during deserialization (and emitted during serialization).

Reply:
Actually, [it does](https://github.com/Azure/azure-sdk-for-net/pull/10667/files#diff-b83cb3aa77c3b3cf010fbe484c6a0198R129). Should we hide it using Pavel's attributes? Seems like we should since it's only a discriminator. Is this common among other services? I can add a bug to Azure/autorest.csharp if so.